torch/nn/functional.py

```diff
@@ -2767,8 +2779,93 @@ def upsample(input, size=None, scale_factor=None, mode='nearest', align_corners=
     warnings.warn("nn.functional.upsample is deprecated. Use nn.functional.interpolate instead.")
     return interpolate(input, size, scale_factor, mode, align_corners)
 
+@overload  # noqa: F811
+def _interp_output_size(dim, closed_over_args):
+    # type: (int, Tuple[Tensor, Optional[int], Optional[List[float]], Optional[bool]]) -> List[int]
+    pass
+
+@overload  # noqa: F811
+def _interp_output_size(dim, closed_over_args):
+    # type: (int, Tuple[Tensor, Optional[List[int]], Optional[List[float]], Optional[bool]]) -> List[int]
+    pass
+
+@overload  # noqa: F811
+def _interp_output_size(dim, closed_over_args):
+    # type: (int, Tuple[Tensor, Optional[int], Optional[float], Optional[bool]]) -> List[int]
+    pass
+
+@overload  # noqa: F811
+def _interp_output_size(dim, closed_over_args):
+    # type: (int, Tuple[Tensor, Optional[List[int]], Optional[float], Optional[bool]]) -> List[int]
+    pass
+
+def _interp_output_size(dim, closed_over_args):  # noqa: F811
```

Comment:
This function is kind of long, can you add an overload for `_check_size_scale_factor` as well?

Reply:
Because of overload decls, i would have to add 16 lines to separate out these 5 error checking lines, i don't think it's worth it, can do it if you think that's worth it.